src/server/req_runjob.c

```diff
@@ -457,15 +454,15 @@ req_runjob(struct batch_request *preq)
 		/* suffix;  in case qrun 1.short vs 1.short.domain.com   */
 
 		snprintf(fixjid, sizeof(fixjid), "%s", jid);
-		pc = strchr(fixjid, (int)'.');
+		pc = strchr(fixjid, (int) '.');
 		if (pc)
 			*pc = '\0';
-		pc = strchr(parent->ji_qs.ji_jobid, (int)'.');
+		pc = strchr(parent->ji_qs.ji_jobid, (int) '.');
 		if (pc)
-			(void)strcat(fixjid, pc);
+			(void) strcat(fixjid, pc);
 
-		(void)strncpy(pdefr->dr_id, fixjid, PBS_MAXSVRJOBID);
-		pdefr->dr_id[PBS_MAXSVRJOBID] ='\0';
+		(void) strncpy(pdefr->dr_id, fixjid, PBS_MAXSVRJOBID);
```

Comment:
Consider removing the (void)'s.  They don't do anything (not even remove the new gcc warning about not using return values).